device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/ProxiedSSLSocket.java

```diff
@@ -96,7 +96,7 @@ private void doTunnelHandshake(Socket tunnel, String host, int port) throws IOEx
         if (this.proxyUsername != null && this.proxyPassword != null)
         {
             String base64EncodedCredentials = new String(Base64.encodeBase64(String.format("%s:%s", this.proxyUsername, new String(this.proxyPassword)).getBytes(byteEncoding)));
-            proxyConnectMessage += String.format("Proxy-Authorization: Basic %s\r\n", base64EncodedCredentials, TransportUtils.USER_AGENT_STRING);
+            proxyConnectMessage += String.format("Proxy-Authorization: Basic %s\r\nUser-Agent: %s\r\n", base64EncodedCredentials, TransportUtils.USER_AGENT_STRING);
```

Comment:
how did you find this? what is the impact?

Reply:
The warning advised me that the string.formal call had more parameters than "%s" instances, so I noticed we were missing the above

Reply:
No meaningful impact. Proxy connect requests don't require a user agent string. It's mostly just used for logging purposes on the user's proxy